### DISCUSSION TITLE
Total number of generated Test Cases

### DISCUSSION THREAD

**Discussion by MAGeorg** [jtpereyda/boofuzz#630] [Q&A] [answered]
Is it possible to know in advance how many Test Cases will be generated during testing?

**Comment by SR4ven** [comment, marked as answer]
Yes, you can use `Session.num_mutations(1)` to get the number of test cases with a depth of 1. The depth has to be consistent with the one handed to `Session.fuzz(1)`.

Without a depth of 1, combinatorial fuzzing is activated and the number of test cases can not be calculated easily.

**Comment by MAGeorg** [comment]
Thanks for your reply. I'm new to using boofuzz and found the mentioned function but didn't quite understand the purpose of the depth parameter. Let me want to form a package consisting of three blocks: s_byte, s_word, s_random. With the parameter depth=1 in fuzzing, will all blocks change or only the first one will change, while the other two will take the default value?

**Reply by SR4ven** [reply]
Sorry for the long delay.
The depth parameter is part of the combinatorial fuzzing. It sets the depth of the combinatorial tree for the test case generation.
With a depth of 1 all primitives will be fuzzed one after another. The primitives that are currently not being touched will render with their default value.
With a depth greater than one, multiple primitives will be fuzzed at the same time. The greater the depth, the more fuzzed primitives will be combined in a test case. This also leads to many more possible test cases, so with the maximum combinatorial depth you can generate an almost infinite amount of cases.
I hope that helps.

**Reply by MAGeorg** [reply]
Thank you very much for your reply, it helped me figure it out.

**Comment by cottesenLM** [comment]
There is also, session.total_num_mutations which i use in callbacks for cheap progress bar functionality.

**Reply by MAGeorg** [reply]
Thanks, but I meant counting the number of Test Cases in advance)
